src/sonic-yang-models/yang-models/sonic-head.yang

```diff
@@ -0,0 +1,101 @@
+module sonic-head {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-head";
+    prefix sonic-head;
+
+    organization "Linkedin Corporation";
+
+    contact "lnos_coders@linkedin.com";
+
+    description "Head yang Module for SONiC OS";
+
+    revision 2019-07-01 {
+        description "First Revision";
+    }
+
+    typedef ip-family {
+        type enumeration {
```

Comment:
Curious! These are generic ones and standard definitions are expected to be available. If so, why do we need to put custom ones.

Reply:
Libyang performs canonization on ietf ip-prefix, Which is not correct as per Sonic operations. So custom definitions are needed. Also there seems some bugs with canonization https://github.com/CESNET/libyang/issues/755.

Basically we need ip-address with prefix-len, which is treated differently than ip-prefix as per ietf definition.